gensim/matutils.py

```diff
@@ -354,12 +372,14 @@ def unitvec(vec):
         return vec
 
     if isinstance(first, (tuple, list)) and len(first) == 2: # gensim sparse format?
-        length = 1.0 * math.sqrt(sum(val ** 2 for _, val in vec))
-        assert length > 0.0, "sparse documents must not contain any explicit zero entries"
-        if length != 1.0:
-            return [(termid, val / length) for termid, val in vec]
-        else:
-            return list(vec)
+        if norm == 'l1':
+            length = float(sum(abs(val) for _, val in vec))
+            assert length > 0.0, "Document contains all zero entries"
```

Comment:
Sorry I had just added that to make it symmetric with the L2 case. I should be returning the vector unchanged in case of 0 length right? Should I proceed with the change?